soroban-sdk/src/deploy.rs

```diff
@@ -152,7 +152,39 @@ impl Deployer {
     /// The TTL is the number of ledgers between the current ledger and the final ledger the data can still be accessed.
     pub fn extend_ttl(&self, contract_address: Address, threshold: u32, extend_to: u32) {
         self.env
-            .extend_contract_instance_and_code_ttl(
+            .extend_contract_instance_ttl(
```

Comment:
I don't think this implementation should be changed - we anyway need to maintain the old host fn and it's more optimal to make a single call.

Reply:
The implementation inside the host of these two versions are identical. Replacing the call to "extend_contract_instance_and_code_ttl" gives chance that this function can be retired (with `max_supported_protocol_number`) one day, since no new contracts will be linked to it. The extra call to host function is negligible (310 cpu instructions for the dispatch).

Reply:
It's a fair argument, there is contract code size as well as small perf benefits to keeping it. I reverted it back.